api/v1beta1/azuremachine_webhook_test.go

```diff
@@ -794,7 +794,22 @@ func TestAzureMachine_ValidateUpdate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "validTest: Spec.NetworkInterfaces.SubnetName will get populated on the first interface",
```

Comment:
What do you mean by first interface?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/71450a89d9f06b75ec46c87b92402d0451a2f050/api/v1beta1/azuremachine_webhook.go#L195 it was written like that in the comments so I took the line, I agree that "validTest: Spec.NetworkInterfaces.SubnetName will get populated for the new object old subnetName is left blank"

Reply:
Ah ok! That is the defaulting logic and as far as this test case is concerned here the field is immutable and it will throw error. 
Also this is kind of covered by `invalidtest: azuremachine.spec.networkInterfaces is immutable` and not sure if we want to have this test case to verify immutability for `empty` --> `non-empty` 
I do not have string opinions and if we still want to have this test case then we need to reword the test description

Reply:
Just put `invalidtest: updating subnet name from empty to non empty` in the description.